src/utils/formated-name.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+interface IFormattedNameProps {
+  name: string;
+}
+
+export const FormattedName: React.FC<IFormattedNameProps> = ({ name }) => {
```

Comment:
Isso pra mim também é estranho, porque você está usando um componente pra fazer o papel de uma função simples. No fim, você retorna um componente, o `<span>`, como resultado da função. Mas o que aconteceria se você precisasse formatar um `<h1>`? Teria que criar outro componente? Outra props? E o `<h2>`? Outro componente? Outra props? Acho que dá pra ver que isso não escala bem.

Acho que tudo poderia ser substituído por uma função simples:

```suggestion
export const formatName = ({ name }) => {
   return `${name.charAt(0).toUpperCase()}${name.slice(1).toLowerCase()}`
}
```

Assim você retorna apenas a string, e pode usar ela em componentes de qualquer tipo.